src/PotentialSimulation/Painting/Painting.jl

```diff
@@ -35,5 +35,156 @@ get_sub_ind_ranges(p::ConstructiveSolidGeometry.AbstractSurfacePrimitive{T}, gri
 get_sub_ind_ranges(p::ConstructiveSolidGeometry.AbstractSurfacePrimitive{T}, grid::CylindricalGrid{T}) where {N,T} = 
     get_sub_ind_ranges((CylindricalPoint.(ConstructiveSolidGeometry.extreme_points(p)), TicksTuple(grid)))
 
-include("Planar.jl")
-include("Curved.jl")
\ No newline at end of file
+function paint!(pointtypes, potential, face::AbstractSurfacePrimitive{T}, geometry, pot_value, grid::CartesianGrid) where {T}
+    t_idx_r1, t_idx_r2, t_idx_r3 = get_sub_ind_ranges(face, grid)
+    ticks = (grid.axes[1].ticks, grid.axes[2].ticks, grid.axes[3].ticks)
+    eX = CartesianVector{T}(1,0,0);
+    eY = CartesianVector{T}(0,1,0);
+    eZ = CartesianVector{T}(0,0,1);
+    widths_ax1 = diff(get_extended_ticks(grid[1]))
+    widths_ax2 = diff(get_extended_ticks(grid[2]))
+    widths_ax3 = diff(get_extended_ticks(grid[3]))
+    for i2 in t_idx_r2
+        for i1 in t_idx_r1
+            l = ConstructiveSolidGeometry.Line(CartesianPoint{T}(ticks[1][i1], ticks[2][i2], zero(T)), eZ)
+            pts = ConstructiveSolidGeometry.intersection(face, l)
+            for pt in pts
+                i3 = searchsortednearest(ticks[3], pt[3])
+                csgtol = abs(ticks[3][i3] - pt[3])
+                Δw_max = max(
+                    widths_ax1[i1], widths_ax1[i1+1],
+                    widths_ax2[i2], widths_ax2[i2+1],
+                    # widths_ax3[i3], widths_ax3[i3+1]
+                )/4
+                if csgtol > Δw_max
+                    csgtol = Δw_max
+                end
+                if in(pt, geometry, csgtol = csgtol)
+                    pointtypes[i1, i2, i3] = zero(PointType)
+                    potential[i1, i2, i3] = pot_value
+                end
+            end
+        end
+    end 
+    for i3 in t_idx_r3
+        for i1 in t_idx_r1
+            l = ConstructiveSolidGeometry.Line(CartesianPoint{T}(ticks[1][i1], zero(T), ticks[3][i3]), eY)
+            pts = ConstructiveSolidGeometry.intersection(face, l)
+            for pt in pts
+                i2 = searchsortednearest(ticks[2], pt[2])
+                csgtol = abs(ticks[2][i2] - pt[2])
+                Δw_max = max(
+                    widths_ax1[i1], widths_ax1[i1+1],
+                    # widths_ax2[i2], widths_ax2[i2+1],
+                    widths_ax3[i3], widths_ax3[i3+1]
+                )/4
+                if csgtol > Δw_max
+                    csgtol = Δw_max
+                end
```

Comment:
The newest commit chooses the values for `Δw_max` and the handling of `csgtol` like it was done for `AbstractPlanarSurfacePrimitives`. For `AbstractCurvedSurfacePrimtives`,  `Δw_max` was divided by `2` instead of `4` and `csgtol` was **NOT** overwritten if it was greater than `Δw_max`. I cannot judge why the differently bent surface primitives were handled differently before but it seems like the tests fail with the implementation for curved surfaces applied to all surfaces...